agents/agents-features/agents-features-opentelemetry/README.md

```diff
@@ -0,0 +1,264 @@
+# OpenTelemetry support
+
+This guide provides comprehensive instructions on how to use OpenTelemetry with the Koog agentic framework for tracing and monitoring your AI agents.
+
+## Overview
+
+OpenTelemetry is an observability framework that provides tools for generating, collecting, and exporting telemetry data (metrics, logs, and traces) from your applications. The Koog OpenTelemetry feature allows you to instrument your AI agents to collect telemetry data, which can help you:
+
+- Monitor agent performance and behavior
+- Debug issues in complex agent workflows
+- Visualize the execution flow of your agents
+- Track LLM calls and tool usage
+- Analyze agent behavior patterns
+
+Key OpenTelemetry concepts
+
+- **Spans**: Spans represent individual units of work or operations within a distributed trace. They indicate the beginning and end of a specific activity in an application, such as an agent execution, a function call, an LLM call, or a tool call.
+- **Exporters**: Exporters are components responsible for sending the collected telemetry data (spans, metrics, logs) to various backends or destinations.
+- **Collectors**: An OpenTelemetry Collector receives, processes, and exports telemetry data. It acts as an intermediary between your applications and your observability backend.
+
+The OpenTelemetry feature in Koog automatically creates spans for various agent events, including:
+
+- Agent execution start and end
+- Node execution
+- LLM calls
+- Tool calls
+
+## Installation
+
+To use OpenTelemetry with Koog, add the OpenTelemetry feature to your agent:
+
+```kotlin
+val agent = AIAgent(
+    executor = simpleOpenAIExecutor(apiKey),
+    llmModel = OpenAIModels.Chat.GPT4o,
+    systemPrompt = "You are a helpful assistant.",
+    installFeatures = {
+        install(OpenTelemetry) {
+            // Configuration options here
+        }
+    }
+)
+```
+
+## Configuration
+
+### Basic configuration
+
+Here is an example of installing the OpenTelemetry feature with a basic set of configuration items:
+
+```kotlin
+install(OpenTelemetry) {
+    serviceName = "my-agent-service"  // Name of your service
+    serviceVersion = "1.0.0"          // Version of your service
+    addOtlpExporter()                 // Add OTLP exporter with default settings
```

Comment:
Maybe simply call it `addExporter()`, as we are already in the `OpenTelemetry` feature DSL? Or `addDefaultExporter`?